### PR TITLE
Integrate mkosi

### PR DESCRIPTION
**Pull request by lucasdemarchi**

### CODE REVIEW

File: Makefile.am [hunk 1]
```diff
@@ -227,3 +227,10 @@ coverity-sync: mavlink-router-coverity-$(shell git rev-parse --short HEAD 2>/dev
 
 coverity-clean:
 	rm -rf mavlink-router-coverity-*.tar.xz cov-int
+
+# ------------------------------------------------------------------------------
+# mkosi
+# ------------------------------------------------------------------------------
+
+mkosi:
+	mkosi -C $(top_srcdir)/tests/osi --build-sources ../../ -f
```
Comment (edersondisouza) [inline Makefile.am]: Shouldn't you check for `mkosi` on configure?

Comment (lucasdemarchi) [inline Makefile.am]: It's one single command. Failing here or during configure is as bad

Comment (anselmolsm) [inline Makefile.am]: It is not a requirement to build the project itself. Same case of cov-build.

File: tests/osi/mkosi.arch [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+Packages = valgrind
+BuildPackages =
+	automake
+	gcc
+	git
+	make
+	pkg-config
+	python2
+	python2-future
+        autoconf
```
Comment (anselmolsm) [inline tests/osi/mkosi.arch]: < picky > there is an additional whitespace here </ >

Comment (lucasdemarchi) [inline tests/osi/mkosi.arch]: fixed.

**Comment by lucasdemarchi** [comment]
@anselmolsm  The main purpose for this being integrated on the tree is to easily compile and run tests on different distros. I did it for Arch since it's the one with the newer gcc 7 and I'm currently on Fedora 25 with gcc 6.x.

**Comment by lucasdemarchi** [comment]
Thank you, pull request merged! [See changes here.](https://github.com/01org/mavlink-router/compare/7196d3d~2...7196d3d)

**Pull request closed without merging**